PlaywrigthUITests/Tests/WebTableTests.cs

```diff
@@ -32,15 +33,121 @@ public async Task VerifyTableVisible2()
-            await DemoQAWebTablesPage.VerifyPopupVisible();
+            await DemoQAWebTablesPage.VerifyPopupVisible(true);
             await DemoQAWebTablesPage.VerifyFirstNameVisible();
         }
 
         //TODO: automate test cases
         //Check any mandatory field
+        [Test]
+        [Description("Verify mandatory fields")]
+        public async Task VerifyDepartmentIsMandatoryField()
+        {
+            //Arrange
+            await DemoQAWebTablesPage.GoToDemoQaWebTablesPage();
+            await DemoQAWebTablesPage.OpenAddPopup();
+            await DemoQAWebTablesPage.VerifyPopupVisible(true);
+            //Act
+            await DemoQAWebTablesPage.FillFirstName("John");
+            await DemoQAWebTablesPage.FillLastName("Doe");
+            await DemoQAWebTablesPage.FillEmail("jon@doe.com");
+            await DemoQAWebTablesPage.FillAge("33");
+            await DemoQAWebTablesPage.FillSalary("555");
+            await DemoQAWebTablesPage.FillDepartment("");
+            await DemoQAWebTablesPage.ClickSubmit();
+            //Assert
+            Assert.That(await DemoQAWebTablesPage.VerifyPopupVisible(false), "The 'Add' popup is still visible.");
+            await Assertions.Expect(DemoQAWebTablesPage.FirstNameField).ToHaveCSSAsync("border-color", "rgb(40, 167, 69)");
+            await Assertions.Expect(DemoQAWebTablesPage.LastNameField).ToHaveCSSAsync("border-color", "rgb(40, 167, 69)");
+            await Assertions.Expect(DemoQAWebTablesPage.EmailField).ToHaveCSSAsync("border-color", "rgb(40, 167, 69)");
+            await Assertions.Expect(DemoQAWebTablesPage.AgeField).ToHaveCSSAsync("border-color", "rgb(40, 167, 69)");
+            await Assertions.Expect(DemoQAWebTablesPage.SalaryField).ToHaveCSSAsync("border-color", "rgb(40, 167, 69)");
+            await Assertions.Expect(DemoQAWebTablesPage.DepartmentField).ToHaveCSSAsync("border-color", "rgb(220, 53, 69)");
+        }
+
         //Add new row and verify row added
+        [Test]
+        [Description("Add new row and verify row added")]
+        [TestCase("John", "Doe", "30", "jon@doe.com", "10000", "IT")]
+        public async Task AddNewRow(string firstName,string lastName, string age, string email, string salary, string department)
+        {
+            //Arrange
+            await DemoQAWebTablesPage.GoToDemoQaWebTablesPage();
+            await DemoQAWebTablesPage.OpenAddPopup();
+            //Act
+            await DemoQAWebTablesPage.FillFirstName(firstName);
+            await DemoQAWebTablesPage.FillLastName(lastName);
+            await DemoQAWebTablesPage.FillEmail(email);
+            await DemoQAWebTablesPage.FillAge(age);
+            await DemoQAWebTablesPage.FillSalary(salary);
+            await DemoQAWebTablesPage.FillDepartment(department);
+            await DemoQAWebTablesPage.ClickSubmit();
+            //Assert
+            Assert.That(await DemoQAWebTablesPage.VerifyPopupVisible(false), "The 'Add' popup is still visible.");
```

Comment:
I wonder how this works.  DemoQAWebTablesPage.VerifyPopupVisible(false) should return false since popup should not be visible. Does Assert.That(someBool) verify at least something?

Reply:
replaced with RegistrationPopup.WaitForAsync